File: nginx-secure-link/hashlink.pl
#!/usr/bin/env perl

use utf8;
use strict;
use warnings;

#
# Perl implementation to generate hashed urls for nginx' secure_link module.
#

require Digest::MD5;

my $prefix = "http://localhost:9005/p";
my $link = "trade-secrets.txt";
my $secret = "secret";

my $link_hash = Digest::MD5::md5_hex("${link}${secret}");

print("${prefix}/${link_hash}/${link}\n");

File: nginx-secure-link/hashlink.py
#!/usr/bin/env python3
"""
	Python implementation to generate hashed urls for nginx' secure_link module.
"""

import hashlib

prefix = "http://localhost:9005/p"
link = "trade-secrets.txt"
secret = "secret"

link_hash = hashlib.md5("{}{}".format(link, secret).encode()).hexdigest()
print("{}/{}/{}".format(prefix, link_hash, link))
